### PR TITLE
[BUGFIX] Document CreateModule - include some code snippets from EXT:example, setDocHeader corrected

### PR DESCRIPTION
**Pull request by format-gmbh**
Releases: main, 12.4

Update Documentation/ExtensionArchitecture/HowTo/BackendModule/CreateModule.rst

### CODE REVIEW

File: Documentation/ExtensionArchitecture/HowTo/BackendModule/_ModuleConfiguration/_AdminModuleControllerSetUpDocHeader.rst.txt [hunk 1]
```diff
@@ -0,0 +1,28 @@
+..  Generated by https://github.com/TYPO3-Documentation/t3docs-codesnippets
+..  Extracted from T3docs\Examples\Controller\AdminModuleController
+
+..  code-block:: php
+    :caption: Class T3docs\\Examples\\Controller\\AdminModuleController
```
Comment (froemken) [inline Documentation/ExtensionArchitecture/HowTo/BackendModule/_ModuleConfiguration/_AdminModuleControllerSetUpDocHeader.rst.txt:5]: Don't know, if it is needed to escape the backslashes here. In next file the backslash is not needed.

Comment (linawolf) [reply Documentation/ExtensionArchitecture/HowTo/BackendModule/_ModuleConfiguration/_AdminModuleControllerSetUpDocHeader.rst.txt:5]: If you use backslashes they need to be escaped. But usually we use filenames and therefore forward slashes that don't need to be escaped

File: Documentation/ExtensionArchitecture/HowTo/BackendModule/CreateModule.rst [hunk 2]
```diff
@@ -22,24 +22,7 @@ Basic controller
...
-    final class AdminModuleController
-    {
-        public function __construct(
-            protected readonly ModuleTemplateFactory $moduleTemplateFactory,
-            protected readonly IconFactory $iconFactory,
-            // ...
-        ) {
-        }
-    }
+..  include:: _ModuleConfiguration/_AdminModuleControllerConstruct.rst.txt
```
Comment (linawolf) [inline Documentation/ExtensionArchitecture/HowTo/BackendModule/CreateModule.rst:25]:
```suggestion
..  include:: _ModuleConfiguration/_AdminModuleControllerConstruct.rst.txt
    :caption: EXT:examples/Classes/Controller/AdminModuleController.php
```

File: Documentation/ExtensionArchitecture/HowTo/BackendModule/CreateModule.rst [hunk 3]
```diff
@@ -84,91 +67,25 @@ Main entry point
...
-               return $this->passwordAction($moduleTemplate);
-           default:
-               $moduleTemplate->setTitle(
-                   $title,
-                   $languageService->sL('EXT:examples/Resources/Private/Language/AdminModule/locallang.xlf:module.menu.log')
-               );
-               return $this->logAction($moduleTemplate);
-       }
-   }
+..  include:: _ModuleConfiguration/_AdminModuleControllerHandleRequest.rst.txt
```
Comment (linawolf) [inline Documentation/ExtensionArchitecture/HowTo/BackendModule/CreateModule.rst:70]:
```suggestion
..  include:: _ModuleConfiguration/_AdminModuleControllerHandleRequest.rst.txt
    :caption: EXT:examples/Classes/Controller/AdminModuleController.php
```

File: Documentation/ExtensionArchitecture/HowTo/BackendModule/CreateModule.rst [hunk 4]
```diff
@@ -84,91 +67,25 @@ Main entry point
...
-    private function setDocHeader(string $active) {
-        $buttonBar = $this->moduleTemplate->getDocHeaderComponent()->getButtonBar();
-        $list = $buttonBar->makeLinkButton()
-            ->setHref('<uri-builder-path>')
-            ->setTitle('A Title')
-            ->setShowLabelText('Link')
-            ->setIcon($this->iconFactory->getIcon('actions-extension-import', Icon::SIZE_SMALL));
-        $buttonBar->addButton($list, ButtonBar::BUTTON_POSITION_LEFT, 1);
-    }
+..  include:: _ModuleConfiguration/_AdminModuleControllerSetUpDocHeader.rst.txt
```
Comment (linawolf) [inline Documentation/ExtensionArchitecture/HowTo/BackendModule/CreateModule.rst:88]:
```suggestion
..  include:: _ModuleConfiguration/_AdminModuleControllerSetUpDocHeader.rst.txt
    :caption: EXT:examples/Classes/Controller/AdminModuleController.php
```

File: Documentation/ExtensionArchitecture/HowTo/BackendModule/CreateModule.rst [hunk 5]
```diff
@@ -177,20 +94,7 @@ and :php:`makeLinkButton()` to create the button. Finally, use :php:`addButton()
...
-   <f:layout name="Module" />
-
-   <f:section name="Content">
-      <h1><f:translate key="function_debug" extensionName="examples"/></h1>
-      <p><f:translate key="function_debug_intro" extensionName="examples"/></p>
-      <p><f:debug inline="1">{cookies}</f:debug></p>
-   </f:section>
-   </html>
-
+..  include:: _ModuleConfiguration/_DebugHtml.rst.txt
```
Comment (linawolf) [inline Documentation/ExtensionArchitecture/HowTo/BackendModule/CreateModule.rst:97]:
```suggestion
..  include:: _ModuleConfiguration/_DebugHtml.rst.txt
    :caption: EXT:examples/Resources/Private/Templates/AdminModule/Debug.html
```

**Review by linawolf — APPROVED** [review]

**Pull request merged**